File: AD-HOC/1271 - Onde estão meus Genes.py
def inverter(genoma: list, inicio: int, fim: int) -> list:
    aux = genoma[inicio:(fim + 1)][::-1]
    return genoma[:inicio] + aux + genoma[(fim + 1):]


cont = 1
while True:
    N = int(input())
    if N == 0:
        break

    genoma = [(i + 1) for i in range(N)]

    R = int(input())
    for i in range(R):
        I, J = list(map(int, input().split()))
        genoma = inverter(genoma, I - 1, J - 1)
    Q = int(input())

    print(f'Genome {cont}')
    for i in range(Q):
        X = int(input())
        print(genoma.index(X) + 1)
    cont += 1

File: AD-HOC/1271 - Onde estão meus Genes.cpp
#include <iostream>

using namespace std;

inline void trocar(int *vetor, int ini, int fim) {
    int meio = (fim - ini + 1) / 2;

    for(int i = 0; i < meio; i++) {
        int aux = vetor[ini + i];
        vetor[ini + i] = vetor[fim - i];
        vetor[fim - i] = aux;
    }

}

int main() {

    int N, R, I, J, Q, X, cont = 1;
    int *genoma;

    while(cin >> N && N != 0) {
        cin >> R;

        genoma = new int[N];
        for(int i = 0; i < N; i++)
            genoma[i] = i + 1;

        for(int i = 0; i < R; i++) {
            cin >> I >> J;
            trocar(genoma, I - 1, J - 1);
        }

        cin >> Q;

        cout << "Genome " << cont << endl;
        for(int i = 0; i < Q; i++) {
            cin >> X;
            for(int i = 0; i < N; i++) {
                if(genoma[i] == X) {
                    cout << i + 1 << endl;
                    break;
                }
            }
        }

        cont++;
        delete[] genoma;

    }

    return 0;
}